fix(web): Inject latest Clerk JWT via Axios interceptor

SokratisVidros opened the pull request:
### What changed? Why was the change needed?

Fetching the Clerk JWT should always be done via the Clerk SDK, not the __session cookie that acts as a cache and creates the risk of getting an outdated JWT.

### Screenshots
![image (1)](https://github.com/user-attachments/assets/5296a719-009c-452f-9b0a-43a5ce51eebe)

SokratisVidros committed:
fix(web): Inject latest Clerk JWT via Axios interceptor

Fetching the Clerk JWT should always be done via the Clerk SDK, not the __session cookie that acts as a cache and creates the risk of getting an outdated JWT.

ChmaraX commented on apps/web/src/api/api.client.ts:
```diff
@@ -78,46 +59,42 @@ function buildUrl(url: string, absoluteUrl: boolean) {
 export function buildApiHttpClient({
   baseURL = API_ROOT || 'https://api.novu.co',
   secretKey,
   jwt,
```
is this `jwt` param needed here? I don't see it being used in the function

SokratisVidros replied:
Good catch. I removed it.

LetItRock commented on apps/web/src/components/providers/AuthProvider.tsx:
```diff
@@ -60,14 +58,10 @@ export const useAuth = () => {
-  const token = getToken();
+export async function getToken() {
+  if (IS_EE_AUTH_ENABLED) {
+    return (await window?.Clerk?.session?.getToken()) || '';
```
Somehow, for me, in my PR `window. Clerk` was a little buggy with probably race-conditons. But when used without the `window` object, it was working fine.

SokratisVidros replied:
Weird, I guess it has to do with React renderings.

SokratisVidros committed:
fix(web): Manage default locale from the API

This change is necessary as the Auth provider was trying to fetch default locale before auth context was resolving from Clerk. So the request fails.

SokratisVidros committed:
Update RootNavMenu.tsx
